src/paulscode/android/mupen64plusae/input/map/PlayerMap.java

```diff
@@ -79,7 +88,9 @@ public String getDeviceSummary( Context context, int player )
             if( mMap.valueAt( i ) == player )
             {
                 int deviceId = mMap.keyAt( i );
-                String name = AbstractProvider.getHardwareName( deviceId );
+                String name = AbstractProvider.isHardwareAvailable( deviceId ) ?
+                        AbstractProvider.getHardwareName( deviceId ) :
+                        "Not Connected";
```

Comment:
I can see why I have more than 1 controller showing up. It's the old saved controller that used to go by device id instead of device descriptor.

To fix this, all you have to do is modify preferences_game.xml. Rename the key "playerMap" to "playerMap1".